plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/ProgramConverter.kt

```diff
@@ -47,19 +49,18 @@ class ProgramConverter(val session: FirSession, override val config: PluginConfi
         return processFunction(symbol, null)
     }
 
-    override fun embedClass(symbol: FirRegularClassSymbol): ClassEmbedding {
-
-        val className = ClassName(symbol.classId.packageFqName, symbol.classId.shortClassName)
-        // If the class name is not contained in the classes hashmap, then add a new embedding.
-        return classes.getOrPut(className) {
-            // Get classes fields
-            val concreteFields = symbol.declarationSymbols
-                .filterIsInstance<FirPropertySymbol>()
-                .filter { it.hasBackingField }
-                .map { VariableEmbedding(it.callableId.embedName(), embedType(it.resolvedReturnType)) }
+    private fun embedClass(symbol: FirRegularClassSymbol): ClassEmbedding {
+        val className = symbol.classId.embedName()
+        var newlyAdded = false
+        val embedding = classes.getOrPut(className) {
+            newlyAdded = true
+            ClassEmbedding(className)
+        }
 
-            ClassEmbedding(className, concreteFields)
+        if (newlyAdded) {
+            processClass(symbol)
         }
```

Comment:
I think instead of the `newlyAdded` boolean I would prefer a control flow like
```kotlin
if (classes.contains(className)) {
  ...
} else {
  ...
}
```
What do you think?

Reply:
It pains me to see it, since it always does two lookups :')  I'll take a look at what Kotlin offers (and it probably is a better idea, really).